### PR TITLE
React native getting started guide updates

### PR DESCRIPTION
**Pull request by abdallahshaban557**
_Issue #, if available:_

_Description of changes:_

React native getting started guide updates

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/fragments/lib/in-app-messaging/js/getting-started.mdx [hunk 1]
```diff
@@ -8,7 +8,7 @@ Make sure you have completed the below steps:
 In your application directory, you should first install the necessary dependencies. Although Amplify In-App Messaging can be used as a standalone JavaScript library, the UI integration is currently available only for _React Native_. The integrations steps here assume a React Native application as the implementation target.
 
 ```bash
-npm install -E aws-amplify@in-app-messaging aws-amplify-react-native@in-app-messaging amazon-cognito-identity-js @react-native-community/netinfo @react-native-async-storage/async-storage @react-native-picker/picker react-native-get-random-values react-native-url-polyfill
+npm install -E aws-amplify@in-app-messaging aws-amplify-react-native@in-app-messaging amazon-cognito-identity-js @react-native-community/netinfo @react-native-async-storage/async-storage react-native-get-random-values react-native-url-polyfill
```
Comment (chintannp) [inline src/fragments/lib/in-app-messaging/js/getting-started.mdx:11]: Do we need `in-app-messaging` from both `aws-amplify@in-app-messaging` and `aws-amplify-react-native@in-app-messaging`?  Additionally, as we are planning to deprecate `aws-amplify-react-native` shouldn't this be removed at all?  @cshfang can you take a look at this?

File: src/fragments/start/getting-started/reactnative/integrate.mdx [hunk 2]
```diff
@@ -347,13 +358,19 @@ Update the `addTodo()` function in the **AddTodoModal** component.
 
 If you try to add todos using the form now, it should successfully close the form when pressing the *Save* button. But our Todo list is still empty even if you restart the app! After initializing our todos as an empty list, we aren't currently updating it again. We will remedy that in the next step.
 
+<Callout>
+
+You will see a console error when you run your app "Datastore - Data won't synchronize", this is expected and will be fixed in a future step once you connect your app to the cloud.
```
Comment (chintannp) [inline src/fragments/start/getting-started/reactnative/integrate.mdx:363]: Excellent Callout! @abdallahshaban557

**Review by chintannp — COMMENTED** [review]
Looks great overall. Just had one question regarding `in-app-messaging` that I asked above.

File: src/fragments/lib/in-app-messaging/js/getting-started.mdx (hunk 1 above)
Comment (abdallahshaban557) [reply src/fragments/lib/in-app-messaging/js/getting-started.mdx:11]: oh - I hadn't really touched that as a part of this PR - but I can remove the picker at least, and the in-app messaging docs will have to change when we have in-app going GA. What do you think @cshfang ?

File: src/fragments/ui-legacy/auth/react/tutorial.mdx [hunk 3]
```diff
@@ -60,7 +60,7 @@ yarn add aws-amplify aws-amplify-react
 If integrating with a React Native app, use:
 
 ```
-yarn add aws-amplify aws-amplify-react-native @react-native-picker/picker
+yarn add aws-amplify aws-amplify-react-native
```
Comment (cshfang) [inline src/fragments/ui-legacy/auth/react/tutorial.mdx:63]: What is ui-legacy? Why do we leave aws-amplify-react-native here but not in the overview file?

File: src/fragments/lib/in-app-messaging/js/getting-started.mdx (hunk 1 above)
Comment (cshfang) [reply src/fragments/lib/in-app-messaging/js/getting-started.mdx:11]: We can leave it for now. But if using `aws-amplify-react-native` for anything other than in-app-messaging, the picker might be needed.

Comment (abdallahshaban557) [reply src/fragments/lib/in-app-messaging/js/getting-started.mdx:11]: ok - let us leave it then. Anything else you need to approve this PR @chintannp ?

File: src/fragments/ui-legacy/auth/react/tutorial.mdx (hunk 3 above)
Comment (rachelnabors) [reply src/fragments/ui-legacy/auth/react/tutorial.mdx:63]: It's still there, needs re-routing

File: src/fragments/lib/auth/js/react-native-withoauth.mdx [hunk 4]
```diff
@@ -21,17 +21,17 @@ The following `props` are used for building a custom UI with buttons if you do n
 To use `withOAuth` in your React Native application first install the appropriate dependencies:
 
 <BlockSwitcher>
-<Block name="Expo">
+<Block name="Expo CLI">
```
Comment (rachelnabors) [inline src/fragments/lib/auth/js/react-native-withoauth.mdx:24]: @abdallahshaban557 Is it "Expo CLI" now and not "Expo?"

**Review by rachelnabors — CHANGES_REQUESTED** [review]

**Review by rachelnabors — APPROVED** [review]

File: src/fragments/lib/datastore/js/getting-started/30_platformIntegration.mdx [hunk 5]
```diff
@@ -14,4 +14,72 @@ Start with [Create React app](https://create-react-app.dev):
 npx create-react-app amplify-datastore --use-npm
 cd amplify-datastore
 npx amplify-app@latest
+```  
+
+</Block>
```
Comment (timngyn) [inline src/fragments/lib/datastore/js/getting-started/30_platformIntegration.mdx:19]: Is this a typo? There isn't an opening block statement for it, or maybe it just needs to be deleted?

**Review by timngyn — CHANGES_REQUESTED** [review]

File: src/fragments/lib/datastore/js/getting-started/30_platformIntegration.mdx (hunk 5 above)
Comment (abdallahshaban557) [reply src/fragments/lib/datastore/js/getting-started/30_platformIntegration.mdx:19]: @timngyn - please do! I think it is a typo for sure

Comment (rachelnabors) [reply src/fragments/lib/datastore/js/getting-started/30_platformIntegration.mdx:19]: I've had a look, and it seems like there's an open tag without a closing tag as well. @abdallahshaban557, mind having a look? I want to make sure we aren't scrobbling this up

**Review by rachelnabors — CHANGES_REQUESTED** [review]

File: src/fragments/lib/datastore/js/getting-started/30_platformIntegration.mdx (hunk 5 above)
Comment (abdallahshaban557) [reply src/fragments/lib/datastore/js/getting-started/30_platformIntegration.mdx:19]: That one can also be removed - I'll submit an update shortly

Comment (abdallahshaban557) [reply src/fragments/lib/datastore/js/getting-started/30_platformIntegration.mdx]: Updated now

**Review by timngyn — APPROVED** [review]

**Review by rachelnabors — APPROVED** [review]

**Pull request merged**